packages/scope-manager/tests/fixtures/class/emit-metadata/accessor-deco.ts.shot

```diff
@@ -0,0 +1,470 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`class emit-metadata accessor-deco 1`] = `
+ScopeManager {
+  variables: Array [
+    ImplicitGlobalConstTypeVariable,
+    Variable$2 {
+      defs: Array [
+        FunctionNameDefinition$1 {
+          name: Identifier<"deco">,
+          node: FunctionDeclaration$1,
+        },
+      ],
+      name: "deco",
+      references: Array [
+        Reference$3 {
+          identifier: Identifier<"deco">,
+          isRead: true,
+          isTypeReference: false,
+          isValueReference: true,
+          isWrite: false,
+          resolved: Variable$2,
+        },
+        Reference$5 {
+          identifier: Identifier<"deco">,
+          isRead: true,
+          isTypeReference: false,
+          isValueReference: true,
+          isWrite: false,
+          resolved: Variable$2,
+        },
+        Reference$7 {
+          identifier: Identifier<"deco">,
+          isRead: true,
+          isTypeReference: false,
+          isValueReference: true,
+          isWrite: false,
+          resolved: Variable$2,
+        },
+        Reference$11 {
+          identifier: Identifier<"deco">,
+          isRead: true,
+          isTypeReference: false,
+          isValueReference: true,
+          isWrite: false,
+          resolved: Variable$2,
+        },
+      ],
+      isValueVariable: true,
+      isTypeVariable: false,
+    },
+    Variable$3 {
+      defs: Array [],
+      name: "arguments",
+      references: Array [],
+      isValueVariable: true,
+      isTypeVariable: true,
+    },
+    Variable$4 {
+      defs: Array [
+        ParameterDefinition$2 {
+          name: Identifier<"param">,
+          node: FunctionDeclaration$1,
+        },
+      ],
+      name: "param",
+      references: Array [],
+      isValueVariable: true,
+      isTypeVariable: false,
+    },
+    Variable$5 {
+      defs: Array [
+        ClassNameDefinition$3 {
+          name: Identifier<"T">,
+          node: ClassDeclaration$2,
+        },
+      ],
+      name: "T",
+      references: Array [
+        Reference$2 {
+          identifier: Identifier<"T">,
+          isRead: true,
+          isTypeReference: true,
+          isValueReference: true,
+          isWrite: false,
+          resolved: Variable$5,
+        },
+        Reference$4 {
+          identifier: Identifier<"T">,
+          isRead: true,
+          isTypeReference: true,
+          isValueReference: true,
+          isWrite: false,
+          resolved: Variable$5,
+        },
+        Reference$6 {
+          identifier: Identifier<"T">,
+          isRead: true,
+          isTypeReference: true,
+          isValueReference: true,
+          isWrite: false,
+          resolved: Variable$5,
+        },
+        Reference$9 {
+          identifier: Identifier<"T">,
+          isRead: true,
+          isTypeReference: true,
+          isValueReference: false,
+          isWrite: false,
+          resolved: Variable$5,
+        },
```

Comment:
From the looks of it - this one is wrong.

This reference is for the pair:
```ts
  set [keyName](a: T) {}
  @deco
  get [keyName]() {}
```

TS will emit [the following code](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAEwKYTgCgHS4A4CGATgQLYBciBYAngJSIDeAvgLABQHEANgQM59EAFSZt2GMHyiIA1qhoA5MqkQBeRAHJgcOBoDcHLrwGIAgkw6JEfVNIDacxcoC6mApSEMWlxAAE0GD4A5raIDvJKpKiuXmLMQA):
```ts
__decorate([
    deco,
    __metadata("design:type", T),
    __metadata("design:paramtypes", [])
], A.prototype, _a, null);
```

So whilst it hasn't referenced the param type - it did reference the return type.
I don't think it's worth spending any time fixing this - it's a super rare usecase. Also it'll be a pretty hard case to reliably fix I think.

I'm 100% okay with just marking this as a KP and only looking at fixing it if someone runs into the issue.